### PR TITLE
Fix links in javadoc and documentations

### PR DESCRIPTION
**Pull request by marcwrobel**
- fix broken links (using their new URL or an alternative URL),
- remove the link to https://quarkus.io/guides/mqtt for now (this guide does not exist),
- use direct links instead of redirects for some links.

A lot of trailing whitespaces has been removed in the process. This should not be a problem so I kept those modifications.

### CODE REVIEW

**Comment by marcwrobel** [comment]
I have also found those broken links in the documentation but I do not know how to fix them :

- https://camel.apache.org/camel-quarkus/latest/reference/extensions/ahc.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/ahc-ws.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/atomix.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/beanio.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/beanstalk.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/elsql.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/etcd.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/ganglia.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/hystrix.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/ipfs.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/jing.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/kamelet-reify.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/msv.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/nagios.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/nsq.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/ribbon.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/sip.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/soroush.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/tagsoup.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/vertx-kafka.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/weka.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/yammer.html
- https://quarkus.io/guides/mqtt
- https://github.com/quarkus/quarkus-quickstarts/tree/main/awt-graphics-rest-quickstart
- https://github.com/quarkus-project/quarkus/issues/120

**Comment by gsmet** [comment]
Thanks. I'll have a look tomorrow.

**Comment by gsmet** [comment]
I force pushed a formatting fix.

**Comment by marcwrobel** [comment]
@gsmet, I'll resolve the conflict.

**Comment by gsmet** [comment]
I grepped the doc and I only find one pointing to https://camel.apache.org/camel-quarkus/latest/, that should be changed to https://camel.apache.org/camel-quarkus/ . The others are in a registry JSON file so they can be simply ignored.

As for MQTT, I would just remove the link for now.

The AWT link should be https://github.com/quarkusio/quarkus-quickstarts/tree/main/awt-graphics-rest-quickstart (note the org is `quarkusio`, not `quarkus`).

https://github.com/quarkus-project/quarkus/issues/120 should be https://github.com/quarkusio/quarkus/issues/120 . It was an early issue and the org name was still in flux :).

Thanks!

**Comment by marcwrobel** [comment]
I refined the script I used and I detected a new broken link : http://escoffier.me. @cescoffier if you want to replace this URL let me know.

**Comment by marcwrobel** [comment]
@gsmet all done.

**Comment by gsmet** [comment]
I think I found it: https://www.escoffier-huissier.com/ . Or maybe not :).

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 1]
```diff
@@ -136,7 +136,7 @@ quarkus.arc.exclude-dependency.acme.artifact-id=acme-services <2>
 == Native Executables and Private Members
 
 Quarkus is using GraalVM to build a native executable.
-One of the limitations of GraalVM is the usage of https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/Limitations.md#reflection[Reflection, window="_blank"].
+One of the limitations of GraalVM is the usage of https://www.graalvm.org/21.3/reference-manual/native-image/Reflection/[Reflection, window="_blank"].
```
Comment (gsmet) [inline docs/src/main/asciidoc/cdi-reference.adoc:139]: Maybe let's point to the latest version that we will soon use: https://www.graalvm.org/22.2/reference-manual/native-image/dynamic-features/Reflection/ .
I don't think we need to upgrade for every version as it's about concepts but better start with the latest.

File: docs/src/main/asciidoc/elasticsearch.adoc [hunk 2]
```diff
@@ -319,7 +319,7 @@ public class SSLContextConfigurator implements RestClientBuilder.HttpClientConfi
     }
 }
 ----
-See https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_encrypted_communication.html[Elasticsearch documentation] for more details on this particular example.
+See https://www.elastic.co/guide/en/elasticsearch/client/java-rest/6.8/_encrypted_communication.html[Elasticsearch documentation] for more details on this particular example.
```
Comment (gsmet) [inline docs/src/main/asciidoc/elasticsearch.adoc:322]: I would go with https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/_encrypted_communication.html instead.

File: docs/src/main/asciidoc/native-reference.adoc [hunk 3]
```diff
@@ -1616,7 +1616,7 @@ Once the image is compiled, enable and start JFR via runtime flags: `-XX:+Flight
     -XX:StartFlightRecording="filename=recording.jfr"
 ----
 
-For more details on using JFR, see https://www.graalvm.org/reference-manual/native-image/JFR[here].
+For more details on using JFR, see https://www.graalvm.org/21.3/reference-manual/native-image/JFR[here].
```
Comment (gsmet) [inline docs/src/main/asciidoc/native-reference.adoc:1619]: I wonder if we will need to put the root in an attribute and update it regularly. But I'm unsure the URL are stable.

We are already using a more recent version and moving to 22.2 soon (same comments for all the GraalVM links, there are a few others).

File: docs/src/main/asciidoc/resteasy-reactive.adoc [hunk 4]
```diff
@@ -9,7 +9,7 @@ include::./attributes.adoc[]
 :jaxrsapi: https://javadoc.io/doc/javax.ws.rs/javax.ws.rs-api/2.1.1
 :jaxrsspec: /specs/jaxrs/2.1/index.html
 :jdkapi: https://docs.oracle.com/en/java/javase/11/docs/api/java.base
-:mutinyapi: https://smallrye.io/smallrye-mutiny/apidocs
+:mutinyapi: https://javadoc.io/doc/io.smallrye.reactive/mutiny/1.6.0
```
Comment (gsmet) [inline docs/src/main/asciidoc/resteasy-reactive.adoc:12]: I would go with this one: https://javadoc.io/doc/io.smallrye.reactive/mutiny/latest/io.smallrye.mutiny/

We usually track the latest or close and we will avoid having to update the versions.

File: docs/src/main/asciidoc/vertx-reference.adoc [hunk 5]
```diff
@@ -103,10 +103,6 @@ Refer to the associated documentation to learn how to use them.
 |`io.quarkus:quarkus-mailer` (extension)
 |https://quarkus.io/guides/mailer
 
-|MQTT Client
-|`io.quarkus:quarkus-smallrye-reactive-messaging-mqtt` (extension)
-|https://quarkus.io/guides/mqtt
```
Comment (gsmet) [inline docs/src/main/asciidoc/vertx-reference.adoc:108]: Ah, I wasn't clear: keep the entry and just remove the link. Better than not having it.

File: extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/ElasticsearchConfig.java [hunk 6]
```diff
@@ -69,7 +69,7 @@ public class ElasticsearchConfig {
      * Thread counts higher than the number of processors should not be necessary because the I/O threads rely on non-blocking
      * operations, but you may want to use a thread count lower than the number of processors.
      *
-     * @see <a href="https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_number_of_threads.html">number of
+     * @see <a href="https://www.elastic.co/guide/en/elasticsearch/client/java-rest/6.8/_number_of_threads.html">number of
```
Comment (gsmet) [inline extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/ElasticsearchConfig.java:72]: https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/_number_of_threads.html

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks for this! I added a few suggestions.

File: docs/src/main/asciidoc/cdi-reference.adoc (hunk 1 above)
Comment (marcwrobel) [reply docs/src/main/asciidoc/cdi-reference.adoc:139]: Ok, done.

File: extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/ElasticsearchConfig.java (hunk 6 above)
Comment (marcwrobel) [reply extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/ElasticsearchConfig.java:72]: Done.

File: docs/src/main/asciidoc/resteasy-reactive.adoc (hunk 4 above)
Comment (marcwrobel) [reply docs/src/main/asciidoc/resteasy-reactive.adoc:12]: It links to the 2.0.0-m1, are you sure ?

File: docs/src/main/asciidoc/native-reference.adoc (hunk 3 above)
Comment (marcwrobel) [reply docs/src/main/asciidoc/native-reference.adoc:1619]: Unfortunately URLs are not stable. This link is not valid for 22.2, it has been moved to https://www.graalvm.org/22.2/reference-manual/native-image/debugging-and-diagnostics/JFR/. I think it's best to keep the version embedded in URLs and update it from time to time.

For the time being I will just update the link to point to 22.2 doc.

File: docs/src/main/asciidoc/vertx-reference.adoc (hunk 5 above)
Comment (marcwrobel) [reply docs/src/main/asciidoc/vertx-reference.adoc:108]: I replaced the link with `No guide yet` to be explicit as to why there is not link. Is this OK with you ?

File: docs/src/main/asciidoc/resteasy-reactive.adoc (hunk 4 above)
Comment (gsmet) [reply docs/src/main/asciidoc/resteasy-reactive.adoc:12]: Hmmmf. Yeah, not ideal. Let's keep it as it is.

File: docs/src/main/asciidoc/vertx-reference.adoc (hunk 5 above)
Comment (gsmet) [reply docs/src/main/asciidoc/vertx-reference.adoc:108]: Yes.

**Review by gsmet — APPROVED** [review]

**Pull request merged**